### PR TITLE
scan: fix scan null pointer bug.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**
1 add put with version
2 fix scan with version bug
3 add more tests

### CODE REVIEW

**Comment by qiuyesuifeng** [comment]
@c4pt0r PTAL

File: put.go [hunk 1]
```diff
@@ -46,6 +48,14 @@ func (p *Put) AddStringValue(family, column, value string) {
 	p.AddValue([]byte(family), []byte(column), []byte(value))
 }
 
+func (p *Put) AddTimestamp(ts uint64) {
+	if ts == 0 {
+		p.Timestamp = math.MaxInt64
```
Comment (siddontang) [inline put.go]: MaxUint64?

Comment (c4pt0r) [inline put.go]: @siddontang no, it should be MaxInt64 here, follow java client's behavior.

Comment (c4pt0r) [inline put.go]: I'll change the uint64 to int64 in next PRs, but I need to do some tests first.

**Comment by c4pt0r** [comment]
LGTM

**Pull request merged**